lib/metadata/util/win32/Win32Software.rb

```diff
@@ -251,11 +251,13 @@ def convert_times(attrs)
     end
 
     def self.DecodeProductKey(product_key)
-      return if product_key.blank? || product_key.length < 67
-      y = []; product_key.split(",")[52..67].each { |b| y << b.hex }
-      return MIQEncode.base24Decode(y)
+      parts = product_key.to_s.split(",")
+      return if parts.length < 67
+
+      ManageIQ::Smartstate::Util.base24_decode(parts[52..67].map(&:hex))
     rescue => err
       $log.error "MIQ(DecodeProductKey): [#{err}]"
+      nil
```

Comment:
This is added because otherwise when a failure occurs the method returns `true` 😕 .  I couldn't figure out how to write a test for this though because I don't think it can fail 🤷‍♂️ .  I only noticed because I forgot to require manageiq/smartstate/util during my development and it was blowing up but my specs were returning `true` 😆